### PR TITLE
reduce duplication and remove dependency to base64

### PR DESCRIPTION
**Pull request by deepu105**
### Summary

I took some inspiration from this project for https://github.com/jwt-rs/jwt-ui, I made some improvements. So contributing that back if there is interest.

- Support JWKS secret for all supported algorithms
- Reduce duplication 
- Remove unwanted base64 dependency
- Move utils to seperate file

### Preflight checklist
- [x] Code formatted rustfmt (`$ cargo fmt`)
- [x] Code linter check with clippy (`$ cargo clippy`)
- [x] Relevant tests added
- [x] Any new documentation added

### CODE REVIEW

File: src/cli_config.rs [hunk 1]
```diff
@@ -123,9 +123,8 @@ pub struct DecodeArgs {
     /// The algorithm used to sign the JWT
     #[clap(long = "alg", short = 'A')]
     #[clap(value_enum)]
-    #[clap(default_value = "HS256")]
     #[clap(value_parser)]
-    pub algorithm: SupportedAlgorithms,
+    pub algorithm: Option<SupportedAlgorithms>,
```
Comment (mike-engel) [inline src/cli_config.rs:127]: I'm not sure what the purpose of this change is. Can you elaborate on why this is needed?

Comment (deepu105) [reply src/cli_config.rs:127]: The algorithm used for the JWT can be derived from the JWT header so making this optional would be better UX IMO

**Comment by mike-engel** [comment]
@deepu105 sorry for the silence on this. Can you rebase/update your branch with `main`?

**Comment by deepu105** [comment]
done

**Review by mike-engel — APPROVED** [review]

**Pull request merged**